sdk/core/System.ClientModel/src/Internal/ChangeTrackingStringList.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace System.ClientModel.Internal;
+
+internal class ChangeTrackingStringList : IList<string>
+{
+    private IList<string> _list;
+    private bool _frozen = false;
+    private bool _tracking = false;
+
+    public ChangeTrackingStringList()
+    {
+        _list = [];
+    }
+
+    public ChangeTrackingStringList(IEnumerable<string> collection)
+    {
+        _list = new List<string>(collection);
+    }
+
+    public void StartTracking() => _tracking = true;
+    public void StopTracking() => _tracking = false;
```

Comment:
Just curious - when do we need these methods? Are there scenarios where we want to construct a list and not track it immediately? Are there scenarios where we want to stop tracking?

Reply:
We don't have scenarios for that atm. This is kind of just following patterns I saw but I think it's a good idea to leave it out and only add it if there are scenarios where it's needed.